src/excmds.ts

```diff
@@ -5899,6 +5899,66 @@ export async function issue() {
     textarea.value = template
 }
 
+/**
+ * Generates qr code for the given text. By default opens in new tab.
+ * If no text is passed as argument then it checks if any selection is present and creates qr code for that.
+ * If not selection is found then it creates qr code for the current tab's URL
+ *
+ * `text2qr --popup [...]` will open the qr code in a new popup window
+ *
+ * `text2qr --window [...]`  will open the qr code in a new window
+ *
+ * `text2qr --current [...]` will open in current tabopen
+ *
+ * `text2qr --timeout <timeout in seconds> [...]` closes the tab/window/popup after specified number of seconds
+ *
+ * Example: text2qr --timeout 5 --popup hello world
+ */
+//#content
+export async function text2qr(...args: string[]) {
+    let text: string = null
+    let isParsed = false
+    let openMode = null
+    let timeout = "-1"
+    while (!isParsed) {
+        switch (args[0]) {
+            case "--window":
+                openMode = winopen
+                args.shift()
+                break
+            case "--popup":
+                openMode = (...args) => winopen("-popup", ...args)
+                args.shift()
+                break
+            case "--current":
+                openMode = open
+                args.shift()
+                break
+            case "--timeout":
+                args.shift()
+                timeout = args[0]
+                args.shift()
+                break
+            default:
+                isParsed = true
+                break
+        }
+    }
+
+    if (!openMode) openMode = tabopen // default to new tab if no option provided
+
+    text = args.join(" ").trim()
+    if (!text || text.length == 0) {
+        text = document.getSelection().toString().trim()
```

Comment:
i just added it to so that no composite commands will be required to use this feature. Although i admit that the steps to follow to use this feature is not very intuitive. Should i remove this?

Reply:
yeah, I think it would probably be better without it.

Thanks for all your work on this!

Reply:
addressed in  [#b50cf4a](https://github.com/tridactyl/tridactyl/pull/4728/commits/b50cf4a3c75957b78d06570d54f2d3a9adeab2a4)